memiavl/multitree.go

```diff
@@ -364,6 +372,8 @@ func (t *MultiTree) WriteSnapshot(dir string) error {
 
 	// write the snapshots in parallel
 	g, _ := errgroup.WithContext(context.Background())
+	g.SetLimit(t.snapshotWriterLimit)
```

Comment:
will it be cleaner if we add this as parameter in `WriteSnapshot` method, so we don't need to add a field in `MultiTree`.

Reply:
> will it be cleaner if we add this as parameter of `WriteSnapshot` method, so we don't need to add a field in `MultiTree`.

Fixed